src/dmd/root/filename.d

```diff
@@ -883,11 +882,13 @@ nothrow:
         }
 
         version (Windows)
-            const r = _mkdir(path.toDString);
+            const r = _mkdir(path);
         version (Posix)
         {
             errno = 0;
-            const r = mkdir(path, (7 << 6) | (7 << 3) | 7);
+            char[] ntPath = path.xarraydup;
```

Comment:
There’s a utility function that might be useful here:

https://github.com/dlang/dmd/blob/8392b50ac2dd2e0e87d0fd700b707d3ca731873a/src/dmd/utils.d#L192